model-optimizer/mo/graph/connection.py

```diff
@@ -83,15 +83,45 @@ def set_source(self, port):
 
         if self.graph.stage == 'front':
             scr_node = port.node
+
```

Comment:
Please, write a lot of unit tests to check new functionality in this file. The tests should be written for case when graph is in "front" stage and "middle" stage.
Ideally we should cover each and every if/else statement.